File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int max_water = 0;
        int w, h;
        while (left < right) {
            w = right - left;
            h = min(height[left], height[right]);
            int water = w * h;

            max_water = max(max_water, water);

            if (height[left] < height[right]) {
                left++;
            }
            else {
                right--;
            }
        }

        return max_water;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int answer = 0;
        int left = 0;
        int right = height.length - 1;

        while (left <= right) {
            int shortHeight = Math.min(height[left], height[right]);
            int width = right - left;
            answer = Math.max(answer, shortHeight * width);

            if (height[left] <= height[right]) {
                left++;
            }
            else {
                right--;
            }
        }

        return answer;
    }
}